lms/djangoapps/courseware/utils.py

```diff
@@ -32,17 +35,35 @@ def verified_upgrade_deadline_link(user, course=None, course_id=None):
     return EcommerceService().upgrade_url(user, course_id)
 
 
-def verified_upgrade_link_is_valid(enrollment=None):
+def can_show_verified_upgrade(user, enrollment, course=None):
     """
-    Return whether this enrollment can be upgraded.
+    Return whether this user can be shown upgrade message.
 
     Arguments:
+        user (:class:`.AuthUser`): The user from the request.user property
```

Comment:
Will the `user` ever be different than `enrollment.user`? If not, is it possible to remove the `user` parameter?

I know you inherited this function from existing code, so if you'd rather leave it as-is, that makes sense.

Reply:
@kdmccormick Actually, the user must be different from `enrollment.user`. This user must come from `request.user`. Only the user object from request.user have the masquerading settings. `enrollment.user` won't have that and hence it masquerading context won't be considered.